feat: add endpoint to retrieve user inventory

HijoDeHonor opened the pull request:
- Implemented GET endpoint '/Inventory/{IdUsuario}' to retrieve a list of items owned by the user
- Added corresponding route, controller, service, and repository
- Created unit tests for the new endpoint, service, and repository

HijoDeHonor committed:
feat: add endpoint to retrieve user inventory

- Implemented GET endpoint '/Inventory/{IdUsuario}' to retrieve a list of items owned by the user
- Added corresponding route, controller, service, and repository
- Created unit tests for the new endpoint, service, and repository

PiewDev commented on dependencyInjection/di-settings.js:
Will separate this file into modules.
![image](https://github.com/HijoDeHonor/game-store-be/assets/43791992/b145a701-43d6-422e-9570-b007a9445d46)
![image](https://github.com/HijoDeHonor/game-store-be/assets/43791992/25160b85-722c-44f7-af1d-332d9773552e)

PiewDev commented on inventory/inventoryController.js:
```diff
@@ -0,0 +1,19 @@
+    this.get = this.get.bind(this);
+  }
+
+  get = tryCatch(async (req, res) => {
```
Be more specific in the name like: getAllUserItems

PiewDev commented on inventory/inventoryController.js:
```diff
@@ -0,0 +1,19 @@
+
+  get = tryCatch(async (req, res) => {
+    const { id } = req.params;
+    if (!id) {
```
Validate this into service

PiewDev commented on inventory/inventoryRepository.js:
```diff
@@ -0,0 +1,36 @@
+      );
+      if (userRows.length === 0) {
+        throw new DoesNotExistError(DOES_NOT_EXIST, INVENTORY);
+      }
```
You already have the logic to obtain a user implemented. Get the user from the service and pass it as a filter for the inventory.

PiewDev commented on inventory/inventoryService.js:
```diff
@@ -0,0 +1,16 @@
+    this.inventoryRepository = inventoryRepository;
+  }
+
+  getAll = async (id) => {
```
Be more specific in the name like: getAllUserItems(userId) or getAllItemsPerUser

PiewDev commented on inventory/inventoryService.js:
```diff
@@ -0,0 +1,16 @@
+
+  getAll = async (id) => {
+    const items = await this.inventoryRepository.getBy(id);
+    if (!items) {
```
Does this if make any sense?

PiewDev reviewed (changes requested):
there are some changes to make

HijoDeHonor committed:
refactor: - Modularized dependencies for better project structure
- Renamed variables and methods to be more descriptive
- Implemented JWT validation for secure access to endpoints

HijoDeHonor committed:
-adds a test to authenticateJWT

HijoDeHonor committed:
refactor: rename of text constants
